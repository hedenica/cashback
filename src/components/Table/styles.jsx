import styled from 'styled-components';

import backgroundImg from '../../assets/img/header-background.svg';

export const TableContainer = styled.div`
  width: 100%;
  height: 100%;
  border-radius: 10px 0 0 10px;
  background: url(${backgroundImg});
  background-size: cover;

  padding: 0 10px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #f7f7f7;

  h3 {
    margin-bottom: 40px;
    font-size: 25px;
  }

  span {
    margin: 20px auto;
    font-size: 13px;
  }
`;

export const Container = styled.table`
  width: 100%;
  text-align: center;
  letter-spacing: 0.5px;
  border-collapse: collapse;
  font-size: 14px;

  thead {
    font-weight: bold;
  }

  table,
  td,
  th {
    border: 2px solid #f7f7f7;
    padding: 5px;
  }

  svg {
    width: 20px;
    height: 20px;
  }
`;
