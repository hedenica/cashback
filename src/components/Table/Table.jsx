import React from 'react';
import { Percent, DollarSign } from 'react-feather';

import { Container, TableContainer } from './styles';

const Table = () => (
  <TableContainer>
    <h3>Regras de Cashback</h3>
    <Container>
      <thead>
        <tr>
          <td>
            <Percent />
            Percentagem
          </td>
          <td>
            <DollarSign />
            Valores de Compra
          </td>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>5%</td>
          <td>R$50,00 até R$200,00</td>
        </tr>
        <tr>
          <td>10%</td>
          <td>R$200,01 até R$500,00</td>
        </tr>
        <tr>
          <td>15%</td>
          <td>R$500,01</td>
        </tr>
      </tbody>
    </Container>
    <span>* Compras acima de R$501,00 ficam em análise por 2 dias úteis</span>
  </TableContainer>
);

export default Table;
