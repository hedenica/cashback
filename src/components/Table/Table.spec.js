import React from 'react';
import { render } from '@testing-library/react';

import Table from './Table';

describe('<Table/>', () => {
  it('should render the table content', () => {
    const { container } = render (
      <Table />
    )

    expect(container).toMatchSnapshot()
  })
})
