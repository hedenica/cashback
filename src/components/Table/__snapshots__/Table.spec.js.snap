// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table/> should render the table content 1`] = `
<div>
  <div
    class="sc-bdfBwQ hcnFOp"
  >
    <h3>
      Regras de Cashback
    </h3>
    <table
      class="sc-gsTCUz hUQIQL"
    >
      <thead>
        <tr>
          <td>
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="19"
                x2="5"
                y1="5"
                y2="19"
              />
              <circle
                cx="6.5"
                cy="6.5"
                r="2.5"
              />
              <circle
                cx="17.5"
                cy="17.5"
                r="2.5"
              />
            </svg>
            Percentagem
          </td>
          <td>
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="12"
                x2="12"
                y1="1"
                y2="23"
              />
              <path
                d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
              />
            </svg>
            Valores de Compra
          </td>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            5%
          </td>
          <td>
            R$50,00 até R$200,00
          </td>
        </tr>
        <tr>
          <td>
            10%
          </td>
          <td>
            R$200,01 até R$500,00
          </td>
        </tr>
        <tr>
          <td>
            15%
          </td>
          <td>
            R$500,01
          </td>
        </tr>
      </tbody>
    </table>
    <span>
      * Compras acima de R$501,00 ficam em análise por 2 dias úteis
    </span>
  </div>
</div>
`;
